tests/Integration/TransactionIntegrationTest.php

```diff
@@ -48,34 +49,49 @@ public function setupOnceDataProvider(): \Generator
         yield [
             Event::createEvent(),
             true,
-            [],
+            null,
             [],
             null,
         ];
 
+        $event = Event::createEvent();
+        $event->setTransaction('/foo/bar');
+
         yield [
-            Event::createEvent(),
+            $event,
             true,
-            ['transaction' => '/foo/bar'],
+            null,
             [],
             '/foo/bar',
         ];
 
         yield [
             Event::createEvent(),
             true,
-            [],
+            null,
             ['PATH_INFO' => '/foo/bar'],
             '/foo/bar',
         ];
 
+        yield [
+            Event::createEvent(),
+            true,
+            EventHint::fromArray([
+                'extra' => [
+                    'transaction' => '/foo/bar',
+                ],
+            ]),
+            [],
+            '/foo/bar',
+        ];
+
         $event = Event::createEvent();
         $event->setTransaction('/foo/bar');
 
         yield [
             $event,
             true,
-            [],
+            null,
```

Comment:
Something is wrong here, this test case looks like it's repeating the one at [line 57](https://github.com/getsentry/sentry-php/pull/1094/files#diff-9198a26b9171256987d7a092025d8ba6R57) 🤔